### PR TITLE
Update actix, futures and tokio

### PR DESCRIPTION
**Pull request by mikhailOK**

### CODE REVIEW

**Comment by mikhailOK** [comment]
#1884

File: Cargo.toml [hunk 1]
```diff
@@ -37,13 +36,13 @@ members = [
 ]
 
 [dev-dependencies]
-actix = "0.8.2"
+actix = "0.9.0-alpha.2"
```
Comment (bowenwang1996) [inline Cargo.toml]: Should we use a pre-release version as dependency? I think 0.9 is going to be released soon https://github.com/actix/actix/commit/8ad0a991b99b1a03f581c1ce33a2c86c5e2c9dac

Comment (mikhailOK) [reply Cargo.toml]: just got released

File: chain/network/src/peer_manager.rs [hunk 2]
```diff
@@ -345,10 +347,10 @@ impl PeerManagerActor {
                 requests.push(active_peer.addr.send(msg.clone()));
             }
         }
-        future::join_all(requests)
+        future::try_join_all(requests)
             .into_actor(self)
-            .map_err(|e, _, _| error!("Failed sending broadcast message: {}", e))
-            .and_then(|_, _, _| actix::fut::ok(()))
+            .map(|x, _, _| x.map_err(|e| error!("Failed sending broadcast message: {}", e)))
+            .map(|_, _, _| ())
```
Comment (bowenwang1996) [inline chain/network/src/peer_manager.rs]: why is the second map needed? Yo can just use a statement in the first map

File: chain/network/src/peer_manager.rs [hunk 3]
```diff
@@ -425,23 +427,25 @@ impl PeerManagerActor {
...
-                        // if let Some(active_peer) = act.active_peers.get(&peer_id1) {
-                        //     active_peer.addr.do_send(PeerManagerRequest::BanPeer(ReasonForBan::Abusive));
-                        // }
-                    } else if let Some(active_peer) = act.active_peers.get_mut(&peer_id1) {
-                        active_peer.full_peer_info.chain_info = res.chain_info;
-                        active_peer.sent_bytes_per_sec = res.sent_bytes_per_sec;
-                        active_peer.received_bytes_per_sec = res.received_bytes_per_sec;
-                    }
-                    actix::fut::ok(())
+                .map(|result, _, _| result.map_err(|err| error!("Failed sending message: {}", err)))
```
Comment (bowenwang1996) [inline chain/network/src/peer_manager.rs]: this map can be merged with the next map

File: chain/network/src/peer_manager.rs [hunk 4]
```diff
@@ -681,16 +685,36 @@ impl PeerManagerActor {
...
 impl Actor for PeerManagerActor {
     type Context = Context<Self>;
 
     fn started(&mut self, ctx: &mut Self::Context) {
         // Start server if address provided.
         if let Some(server_addr) = self.config.addr {
             // TODO: for now crashes if server didn't start.
-            let listener = TcpListener::bind(&server_addr).unwrap();
+            let listener = std::net::TcpListener::bind(&server_addr).unwrap();
+            let listener = TcpListener::from_std(listener).unwrap();
```
Comment (bowenwang1996) [inline chain/network/src/peer_manager.rs]: why not use `tokio::net::TcpListener` directly?

File: chain/network/src/peer_manager.rs [hunk 5]
```diff
@@ -681,16 +685,36 @@ impl PeerManagerActor {
...
+            let incoming = IncomingCrutch { listener };
```
Comment (bowenwang1996) [inline chain/network/src/peer_manager.rs]: @frol is there a better way to do this?

File: chain/network/src/types.rs [hunk 6]
```diff
@@ -845,12 +847,14 @@ impl InboundTcpConnect {
...
 /// Actor message to request the creation of an outbound TCP connection to a peer.
 #[derive(Message)]
+#[rtype(result = "()")]
 pub struct OutboundTcpConnect {
     /// Peer information of the outbound connection
     pub peer_info: PeerInfo,
 }
 
 #[derive(Message, Clone, Debug)]
+#[rtype(result = "()")]
```
Comment (bowenwang1996) [inline chain/network/src/types.rs]: What is `rtype`?

Comment (mikhailOK) [reply chain/network/src/types.rs]:
```
/// Represent message that can be handled by an actor.
pub trait Message {
    /// The type of value that this message will resolved with if it is
    /// successful.
    type Result: 'static;
}
```
`Result` for actix derive macro, old version allowed no argument to default to `()`

File: chain/network/src/peer_manager.rs (hunk 5 above)
Comment (frol) [reply chain/network/src/peer_manager.rs]: @bowenwang1996 There is not something that I can suggest right now. This does not look terrible to me.

File: core/primitives/src/test_utils.rs [hunk 7]
```diff
@@ -46,13 +47,28 @@ pub fn init_test_module_logger(module: &str) {
...
         .filter(None, LevelFilter::Info)
         .filter(Some("actix_web"), LevelFilter::Warn)
         .try_init();
+    init_stop_on_panic();
+}
+
+static SET_PANIC_HOOK: Once = Once::new();
+
+/// This is a workaround to make actix/tokio runtime stop when a task panics.
+pub fn init_stop_on_panic() {
```
Comment (nearmax) [inline core/primitives/src/test_utils.rs]: Doesn't our node stop when one of the actix tasks panic?

Comment (mikhailOK) [reply core/primitives/src/test_utils.rs]: not with tokio 0.2: now it catches panics

**Pull request merged**